packages/augur-sdk/src/state/db/MarketDB.ts

```diff
@@ -229,6 +230,7 @@ export class MarketDB extends DerivedDB {
   }
 
   protected processDoc(log: ParsedLog): ParsedLog {
+    this.lock(this.PROCESS_DOC_LOCK);
```

Comment:
I think this should actually go in DerivedDB.handleMergeEvent and should be unlocked at the end of it.

Reply:
ah so do it for all of them. will do

Reply:
The important thing is that that will make the lock persist during all of the event handling and only be released once its over. If its in process docs then it will be toggled for each doc in a large set

Reply:
It was only being toggled on, never off, so it would have persisted. No reason to toggle in on a bunch of times though when once will do.